archinstall/lib/disk/partitioning_menu.py

```diff
@@ -296,9 +308,13 @@ def handle_action(
 					self._prompt_formatting(partition)
 				case 'mark_bootable':
 					partition.invert_flag(PartitionFlag.BOOT)
-					if self._using_gpt:
-						partition.invert_flag(PartitionFlag.ESP)
+				case 'mark_esp':
+					if PartitionFlag.XBOOTLDR in partition.flags:
+						partition.invert_flag(PartitionFlag.XBOOTLDR)
+					partition.invert_flag(PartitionFlag.ESP)
 				case 'mark_xbootldr':
+					if PartitionFlag.ESP in partition.flags:
+						partition.invert_flag(PartitionFlag.ESP)
```

Comment:
Could you explain this?

Reply:
Can this instead avoid all invalid combinations? For example, if the SWAP flag was set this would still allow adding the ESP or XBOOTLDR flag without removing the SWAP flag.

Reply:
Done that.